File: java/leetcode/src/main/java/com/leetcode/BST/Minimum_Absolute_Difference_in_BST_530.java
package com.leetcode.BST;

public class Minimum_Absolute_Difference_in_BST_530 {
    int min = Integer.MAX_VALUE;
    TreeNode pre = null;
    public int getMinimumDifference(TreeNode root) {
        dfs(root);
        return min;
    }

    void dfs(TreeNode node) {
        if (node == null) return;
        dfs(node.left);
        if (pre != null) {
            int diff = Math.abs(node.val - pre.val);
            if (diff < min) min = diff;
        }
        pre = node;
        dfs(node.right);
    }
}

File: js/algrithm/BST/Minimum_Absolute_Difference_in_BST_530.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var getMinimumDifference = function(root) {
    let min = Infinity
    let pre = undefined
    const dfs = function(node) {
        if (!node) return
        dfs(node.left)
        if (pre) {
            const abs = Math.abs(pre.val - node.val)
            min = Math.min(min, abs)
        }
        pre = node
        dfs(node.right)
    }
    dfs(root)
    return min
};